src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1677,6 +1707,61 @@ internal int GetTargetAttributeSignatureIndex(CustomAttributeHandle customAttrib
             }
         }
 
+        internal bool GetCustomAttribute(
+            CustomAttributeHandle handle,
+            IMethodSymbolInternal attributeConstructor,
+            out TypedConstant[] positionalArgs,
+            out KeyValuePair<string, TypedConstant>[] namedArgs)
+        {
+            try
+            {
+                positionalArgs = Array.Empty<TypedConstant>();
+                namedArgs = Array.Empty<KeyValuePair<string, TypedConstant>>();
+
+                BlobReader argsReader = Module.GetMemoryReaderOrThrow(Module.GetCustomAttributeValueOrThrow(handle));
+
+                uint prolog = argsReader.ReadUInt16();
+                if (prolog != 1)
+                {
+                    return false;
+                }
+
+                int paramCount = attributeConstructor.ParameterCount;
+
+                if (paramCount > 0)
+                {
+                    positionalArgs = new TypedConstant[paramCount];
+
+                    for (int i = 0; i < positionalArgs.Length; i++)
+                    {
+                        var parameterType = attributeConstructor.GetParameterType(i);
+                        positionalArgs[i] = DecodeCustomAttributeFixedArgumentOrThrow(parameterType, ref argsReader);
+                    }
+                }
+
+                short namedParamCount = argsReader.ReadInt16();
+
+                if (namedParamCount > 0)
+                {
+                    namedArgs = new KeyValuePair<string, TypedConstant>[namedParamCount];
+
+                    for (int i = 0; i < namedArgs.Length; i++)
+                    {
+                        (namedArgs[i], _, _, _) = DecodeCustomAttributeNamedArgumentOrThrow(ref argsReader);
+                    }
+                }
+
+                return true;
+            }
+            catch (Exception e) when (e is UnsupportedSignatureContent || e is BadImageFormatException)
+            {
+                positionalArgs = Array.Empty<TypedConstant>();
+                namedArgs = Array.Empty<KeyValuePair<String, TypedConstant>>();
+            }
+
+            return false;
+        }
+
         internal bool GetCustomAttribute(
```

Comment:
>GetCustomAttribute [](http://example.com/codeflow?start=22&length=18)

It looks like there is no good reason to keep this function, because it is only used to crack a single well-known attribute and that code is an "outlier", i.e. the only well-known attribute cracked in an unconventional way by using ```MetadataDecoder``` instead of having ```PEModule``` do the work.

It is fine to not do the work in context of this PR, but we should realign the handling of the attribute usage attribute and get rid of this function and `DecodeCustomAttributeParameterTypeOrThrow`, because we might get in trouble in the future by using them in context that might need to handle generic attributes or generic types in their application. 

Please open a follow-up issue if you prefer this option.

Reply:
https://github.com/dotnet/roslyn/issues/70338